bridges/modules/xcm-bridge-router/src/lib.rs

```diff
@@ -0,0 +1,444 @@
+		/// The overarching event type.
+		#[allow(deprecated)]
+		#[pallet::no_default_bounds]
+		type RuntimeEvent: From<Event<Self, I>>
+			+ IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// Benchmarks results from runtime we're plugged into.
+		type WeightInfo: WeightInfo;
+
+		/// Checks the XCM version for the destination.
+		type DestinationVersion: GetVersion;
+
+		/// The bridge hub may be:
+		/// - A system (sibling) bridge hub parachain (or another chain), in which case we need an
+		///   implementation for `T::MessageExporter` that sends `ExportMessage`, e.g.,
+		///   `SovereignPaidRemoteExporter`.
+		/// - The local chain, in which case we need an implementation for `T::MessageExporter` that
+		///   does not use `ExportMessage` but instead directly calls the `ExportXcm`
+		///   implementation.
+		#[pallet::no_default]
+		type MessageExporter: SendXcm;
+
+		/// Resolves a specific `BridgeId` for `dest`, used for identifying the bridge in cases of
+		/// congestion and dynamic fees. If it resolves to `None`, it means no congestion or
+		/// dynamic fees are handled for `dest`.
+		#[pallet::no_default]
+		type BridgeIdResolver: ResolveBridgeId;
+
+		/// Origin that is allowed to update bridge status,
+		/// e.g. the sibling bridge hub or governance as root.
+		#[pallet::no_default]
+		type UpdateBridgeStatusOrigin: EnsureOriginWithArg<Self::RuntimeOrigin, BridgeIdOf<Self, I>>;
+
+		/// Additional fee that is paid for every byte of the outbound message.
+		/// See `calculate_message_size_fee` for more details.
+		type ByteFee: Get<u128>;
+		/// Asset used to pay the `ByteFee`.
+		/// If not specified, the `ByteFee` is ignored.
+		/// See `calculate_fees` for more details.
+		type FeeAsset: Get<Option<AssetId>>;
+	}
+
+	/// An alias for the `BridgeId` of configured `T::BridgeIdResolver`.
+	pub type BridgeIdOf<T, I> = <<T as Config<I>>::BridgeIdResolver as ResolveBridgeId>::BridgeId;
+
+	#[pallet::pallet]
+	pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
+
+	#[pallet::call]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Notification about congested bridge queue.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::update_bridge_status())]
+		pub fn update_bridge_status(
+			origin: OriginFor<T>,
+			bridge_id: BridgeIdOf<T, I>,
+			is_congested: bool,
+		) -> DispatchResult {
+			let _ = T::UpdateBridgeStatusOrigin::ensure_origin(origin, &bridge_id)?;
+
+			log::info!(
+				target: LOG_TARGET,
+				"Received bridge status from {:?}: congested = {}",
+				bridge_id,
+				is_congested,
+			);
+
+			// update status
+			Self::do_update_bridge_status(bridge_id, is_congested);
+
+			Ok(())
+		}
+	}
+
+	/// Stores `BridgeState` for congestion control and dynamic fees for each resolved bridge ID
+	/// associated with a destination.
+	#[pallet::storage]
+	pub type Bridges<T: Config<I>, I: 'static = ()> =
+		StorageMap<_, Blake2_128Concat, BridgeIdOf<T, I>, BridgeState, OptionQuery>;
```

Comment:
Could we use a `ValueQuery` here and have the default `BridgeState`:

```
BridgeState { delivery_fee_factor: MINIMAL_DELIVERY_FEE_FACTOR, is_congested: false }
```

?

I think we can consider that we always apply a `delivery_fee_factor` of 1.

Reply:
fixed by @rosarp here: https://github.com/paritytech/polkadot-sdk/pull/9268